src/cc/usdt/usdt.cc

```diff
@@ -295,24 +295,46 @@ Probe *Context::get(const std::string &provider_name,
 
 bool Context::enable_probe(const std::string &probe_name,
                            const std::string &fn_name) {
+  return enable_probe("", probe_name, fn_name);
+}
+
+bool Context::enable_probe(const std::string &provider_name,
+                           const std::string &probe_name,
+                           const std::string &fn_name) {
   if (pid_stat_ && pid_stat_->is_stale())
     return false;
 
-  // FIXME: we may have issues here if the context has two same probes's
-  // but different providers. For example, libc:setjmp and rtld:setjmp,
-  // libc:lll_futex_wait and rtld:lll_futex_wait.
+  unsigned int matches = 0;
   Probe *found_probe = nullptr;
   for (auto &p : probes_) {
     if (p->name_ == probe_name) {
-      if (found_probe != nullptr) {
-         fprintf(stderr, "Two same-name probes (%s) but different providers\n",
-                 probe_name.c_str());
-         return false;
+      if (found_probe == nullptr && provider_name == "")
+      {
+        found_probe = p.get();
+        matches++;
+      }
+      else if (found_probe != nullptr && provider_name == "")
+      {
+        fprintf(stderr, "Found duplicate provider (%s) for underspecified probe (%s)\n",
+                p->provider().c_str(), p->name().c_str());
+        matches++;
+      } else if (provider_name != "" && p->provider() == provider_name)
+      {
+        found_probe = p.get();
+        matches++;
       }
-      found_probe = p.get();
     }
   }
 
+  if (matches > 1) {
+    fprintf(stderr, "Found %i duplicate providers for underpecified probe (%s)\n",
+                matches, fn_name.c_str());
+    return false;
+  } else if(matches < 1) {
```

Comment:
I think simply return `false` would do. Caller are already checking the not found failure result.

Reply:
Yes I'll clean this up, your above suggestion negates the need for this.

Reply:
I just outright got rid of this, as I ended up putting the same thing in my bpftrace PR anyways.

I would expect other users (such as scripts in this repo) should take the same approach of handling this. I'd be happy to help make those changes as well.